verilog/analysis/checkers/token_stream_lint_rule.cc

```diff
@@ -69,7 +70,16 @@ void TokenStreamLintRule::HandleSymbol(const verible::Symbol& symbol,
   verible::matcher::BoundSymbolManager manager;
   if (StringLiteralMatcher().Matches(symbol, &manager)) {
     const auto& string_node = SymbolCastToNode(symbol);
-    const auto& string_literal = SymbolCastToLeaf(*string_node.children()[0]);
+    const auto& node_children = string_node.children();
+    const auto& literal =
```

Comment:
So could it be that we actually can have more than one string literal in this node ?
I suspect in our ternary situation we might have some TK_StringLiteral ':' TK_StringLiteral ?
Here, we're only looking for the first occurence.

So if there are multiple, should we check all of the children's string-literals ?
(might actually simplify the code as we can simply loop through the node-children and `continue` on non-`TK_StringLiteral`-ness).

Reply:
In our ternary example when I printed the tree I found such structure:
```
Node @0 (tag: kConditionExpression) {
   Node @0 (tag: kNumber) {
      Leaf @0 (#TK_DecNumber @27-28: "1")
   }
   Leaf @1 (#'?' @29-30: "?")
   Node @2 (tag: kExpression) {
         Leaf @0 (#TK_StringLiteral @31-36: ""foo"")
   }
   Leaf @3 (#':' @37-38: ":")
   Leaf @4 (#TK_StringLiteral @39-44: ""bar"")
}
```
And as we can see we have two string literals. So the symbols passed to checker are: `kConditionExpression` and `kExpression`.  So for the first node, I'm looking for  `TK_StringLiteral` within its children, and the same is done for the second obtained symbol.
In the previous solution, I assumed that the string literal will be always in the first position in the children's leaf-list.

I believe that the later `if` validation can be removed cause the checker symbol is guaranteed to points to a node with TK_StringLiteral and there is no chance that there wouldn't be the desired string token.

So shouldn't I look for the string literal for each given symbol within the symbol's children? (but only leaves are considered, I do not dive into the nodes)
Do I understand it correctly?